src/python/pants/backend/python/rules/hermetic_pex.py

```diff
@@ -45,24 +46,31 @@ def create_process(
         # execution (so long as `env` is populated with a `PATH` env var and `python` is discoverable
         # somewhere on that PATH). This is only used to run the downloaded PEX tool; it is not
         # necessarily the interpreter that PEX will use to execute the generated .pex file.
+        escaped_pex_args = " ".join(f"'{arg}'" for arg in pex_args)
+        script = dedent(
+            f"""
+            export PEX_ROOT="${{_APPEND_ONLY_CACHE_pex_root:-pex_root}}"
+            export RUNTIME_PEX_ROOT="$PEX_ROOT"
+            python '{pex_path}' {escaped_pex_args}
+            """
+        )
+
         # TODO(#7735): Set --python-setup-interpreter-search-paths differently for the host and target
         # platforms, when we introduce platforms in https://github.com/pantsbuild/pants/issues/7735.
-        argv = ("python", pex_path, *pex_args)
-
         hermetic_env = dict(
             PATH=create_path_env_var(python_setup.interpreter_search_paths),
-            PEX_ROOT="./pex_root",
             PEX_INHERIT_PATH="false",
             PEX_IGNORE_RCFILES="true",
-            **subprocess_encoding_environment.invocation_environment_dict
+            **subprocess_encoding_environment.invocation_environment_dict,
         )
         if env:
             hermetic_env.update(env)
 
         return Process(
-            argv=argv,
+            argv=("/bin/bash", "-c", script),
             input_digest=input_digest,
             description=description,
             env=hermetic_env,
-            **kwargs
+            append_only_caches={"pex_root"},
```

Comment:
Instead of plumbing the actual cache location into the process via an env var, could we just have the expected location be part of the configuration here. IOW:
```python
append_only_caches={"pex_root": "pex_root"},
```
Which says: I'd like the named cache "pex_root" to be linked into my sandbox at the path `./pex_root`. The local process setup could then do just that and use a symlnk to provide the engine controlled cache dir to the process. This avoids the script glue above for users of the API.

Reply:
FWIW, this seems like a sensible remote API as well.

Reply:
I like it :)

Bikeshed: Just a set, and have them implicitly be directories in `.cache` (or `cache`).

So a request with:

`append_only_caches={"pex_root"}`

would create a directory relative to the execution root:

`/root/.caches/pex_root`

Reply:
Mm. Yea, that works: thanks.

Unsure on the dict vs set choice though. Does having a stable location within the workspace assist with Docker usage in some way...?

Reply:
Its just simpler. Now any user of this feature can, in the `Process` request setup the location of the cache they desire and plumb it through to the tool with an env var or an argv and not need to write a glue script.

Reply:
I'm mostly thinking that having a set is defensive for how the upstream standardisation happens. If when we talk to the other remote execution folks, consensus is to go for a dict, it's easy for us to go from set to dict (we could continue to expose a set to rules, or accept both and have a default fallback for sets, or similar). If we do a dict here, and end up specifying a set for the RE stuff, it's a breaking change for us to take away that flexibility we offered rule-writers, and I don't see particular value in the flexibility. If there's a reason for allowing the flexibility, that's a different story :)

A set also slightly avoids needing to spec a few edge cases (e.g. banning one specified cache being a subdir of another), but again, if there's a reason for the flexibility, that's fine.

Reply:
We have no usecases I know of that need the flexibility that I know of. A set + convention sgtm.

Reply:
An issue with the set though, is that the location of the caches would be baked-in and universal in the standard... all processes everywhere would need to be compatible with the chosen location. Unclear whether that is likely to fly?

... having said that, landing this change won't marry us to one or the other. I'll get the conversation started though.